src/app/search-bar/search-bar.component.ts

```diff
@@ -45,7 +54,44 @@ export class SearchBarComponent {
       readerPubKey = this.globalVars.loggedInUser.PublicKeyBase58Check;
     }
 
-    this.backendApi
+    // If we are searching for a public key, call get single profile with the public key.
+    if (
```

Comment:
Do we do these checks anywhere else in the repo? If so, worth making a helper?

Reply:
we have a function called `isMaybePublicKey` :joy: I'll update it to include length checks and use it here.